File: modules/desktop-client-sdk/src/main/java/com/avandocmsg/messenger/desktop/sdk/call/NativeCallAudioMedia.java
package com.avandocmsg.messenger.desktop.sdk.call;

import com.avandocmsg.messenger.media.NativeWebRtcAudioClient;
import java.util.function.Consumer;

public final class NativeCallAudioMedia implements CallAudioMedia {

    private final NativeWebRtcAudioClient client = NativeWebRtcAudioClient.create();

    @Override
    public String createOffer() {
        return client.createOffer();
    }

    @Override
    public void connect(String answerSdp) throws Exception {
        client.connect(answerSdp);
    }

    @Override
    public void sendPcmu(byte[] payload) {
        client.sendPcmu(payload);
    }

    @Override
    public void onPcmu(Consumer<byte[]> listener) {
        client.onPcmu(listener);
    }

    @Override
    public boolean mediaReady() {
        return client.mediaReady();
    }

    @Override
    public void close() {
        client.close();
    }
}

File: mobile/mobile-client-sdk/src/androidMain/kotlin/com/avandocmsg/messenger/mobile/sdk/call/NativeCallAudioMedia.kt
package com.avandocmsg.messenger.mobile.sdk.call

import com.avandocmsg.messenger.media.NativeWebRtcAudioClient
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext

class NativeCallAudioMedia : CallAudioMedia {
    private val client = NativeWebRtcAudioClient.create()

    override fun createOffer(): String = client.createOffer()

    override suspend fun connect(answerSdp: String) {
        withContext(Dispatchers.IO) {
            client.connect(answerSdp)
        }
    }

    override fun sendPcmu(payload: ByteArray) {
        client.sendPcmu(payload)
    }

    override fun onPcmu(listener: (ByteArray) -> Unit) {
        client.onPcmu { payload -> listener(payload) }
    }

    override fun mediaReady(): Boolean = client.mediaReady()

    override fun close() {
        client.close()
    }
}
